go/vt/mysqlctl/mycnf_gen.go

```diff
@@ -131,19 +131,27 @@ func (cnf *Mycnf) fillMycnfTemplate(tmplSrc string) (string, error) {
 
 // RandomizeMysqlServerID generates a random MySQL server_id.
 //
-// The value assigned to ServerID will be in the range [100, 2^32).
-// It avoids 0 because that's reserved for mysqlbinlog dumps.
-// It also avoids 1-99 because low numbers are used for fake slave connections.
-// See NewSlaveConnection() in slave_connection.go for more on that.
+// The value assigned to ServerID will be in the range [100, 2^31):
+// - It avoids 0 because that's reserved for mysqlbinlog dumps.
+// - It also avoids 1-99 because low numbers are used for fake slave
+// connections.  See NewSlaveConnection() in slave_connection.go for
+// more on that.
+// - It avoids the 2^31 - 2^32-1 range, as there seems to be some confusion
+// between the documentation at:
+// http://dev.mysql.com/doc/refman/5.7/en/replication-options.html
+// that implies is a full 32 bits number, and the code that encodes or
+// decodes serverIDs from core mysqld, which uses signed int32.
```

Comment:
I haven't seen any evidence that code in core mysqld uses signed int32. The problem I observed was that the semi-sync plugin appears to use signed int32.